### DISCUSSION TITLE
`evil-normal-state` doesn't work when called from an async callback of plz

### DISCUSSION THREAD

**Discussion by NightMachinery** [alphapapa/plz.el#59] [Q&A] [answered]
It seems `evil-normal-state` doesn't work when called from an async callback of plz. Any ideas?

```
(defun test-plz-1 ()
  "Test command to demonstrate `evil-normal-state` in an async callback of `plz`."
  (interactive)
  (message "Starting test-plz-1...")
  (plz 'get "https://httpbin.org/get"
       :as 'string
       :then (lambda (response)
               (message "Received response: %s" response)
               ;; Attempt to switch to normal state
               (evil-normal-state)
               (message "Attempted to switch to normal state"))))
```

To reproduce:
1. Switch to evil insert state.
2. Call `test-plz-1`.
3. Evil is still in insert state.

**Comment by alphapapa** [comment, marked as answer]
When writing Elisp, you must be aware of which buffer is current when your code is evaluated.  You are writing an asynchronous request, so you must not assume which buffer will be current when the callback is called.  You will need to set the current buffer for your code to have the desired effect.
